state/relation.go

```diff
@@ -608,6 +608,12 @@ func (r *Relation) removeRemoteEndpointOps(ep Endpoint, unitDying bool) ([]txn.O
 
 		app := &RemoteApplication{st: r.st}
 		hasLastRef := bson.D{{"life", Dying}, {"relationcount", 1}}
+		// Cross model relations might get out of sync between models.
+		// If this is not the first attempt to run the remove endpoint txns,
+		// force through the removal so long as the app is not alive.
+		if attempt > 0 {
+			hasLastRef = bson.D{{"life", bson.D{{"$ne", Alive}}}}
```

Comment:
could this assertion be used even on the first attempt?

Reply:
As in, if the first assertion fails because relation count is > 1, we are gonna force remove it anyway in the second attempt only a second later.

Reply:
yeah, can make that change